master/internal/trial.go

```diff
@@ -748,9 +748,10 @@ func (t *trial) transition(s model.StateWithReason) error {
 			})
 		default:
 			if action, ok := map[model.State]task.AllocationSignal{
-				model.StoppingCanceledState: task.TerminateAllocation,
-				model.StoppingKilledState:   task.KillAllocation,
-				model.StoppingErrorState:    task.KillAllocation,
+				model.StoppingCompletedState: task.TerminateAllocation,
+				model.StoppingCanceledState:  task.TerminateAllocation,
+				model.StoppingKilledState:    task.KillAllocation,
+				model.StoppingErrorState:     task.KillAllocation,
```

Comment:
do we expect this list to be different from `model.StoppingStates` ? if not wdy about having a default terminate action and escalate it to kill in those two states? or have some sort of `else` handling in case this if statement doesn't match the state which seems like it was the cause of the issue?

Reply:
refactored